base/libgit2/repository.jl

```diff
@@ -115,14 +120,65 @@ function get{T <: GitObject}(::Type{T}, repo::GitRepo, oid::AbstractString)
     return get(T, repo, GitHash(oid), length(oid))
 end
 
+"""
+    LibGit2.gitdir(repo::GitRepo)
+
+Returns the location of the "git" files of `repo`:
+
+ - for normal repositories, this is the location of the `.git` folder.
+ - for bare repositories, this is the location of the repository itself.
+
+See also `workdir`, `path`
+"""
 function gitdir(repo::GitRepo)
     return unsafe_string(ccall((:git_repository_path, :libgit2), Cstring,
                         (Ptr{Void},), repo.ptr))
 end
 
+"""
+    LibGit2.workdir(repo::GitRepo)
+
+The location of the working directory of `repo`. This will throw an error for bare
+repositories.
+
+!!! note
+
+    This will typically be the parent directory of `gitdir(repo)`, but can be different in
+    some cases: e.g. if either the `core.worktree` configuration variable or the
+    `GIT_WORK_TREE` environmental variable is set.
```

Comment:
we use "environment variable" far more often than "environmental variable," the latter just in a couple of places